docs/source/examples/e2e_flow.rst

```diff
@@ -0,0 +1,519 @@
+
+.. code-block:: yaml
+
+    checkpointer:
+        # we need to use the custom TorchTune checkpointer
+        # instead of the HF checkpointer for loading
+        # quantized models
+        _component_: torchtune.utils.FullModelTorchTuneCheckpointer
+
+        # directory with the checkpoint files
+        # this should match the output_dir specified during
+        # finetuning
+        checkpoint_dir: <checkpoint_dir>
+
+        # checkpoint files point to the quantized model
+        checkpoint_files: [
+            hf_model_0001_0-4w.pt,
+        ]
+
+        output_dir: <checkpoint_dir>
+        model_type: LLAMA2
+
+    # we also need to update the quantizer to what was used during
+    # quantization
+    quantizer:
+        _component_: torchtune.utils.quantization.Int4WeightOnlyQuantizer
+        groupsize: 256
+
+
+Once the config is updated, let's kick off generation! We'll use the
+same sampling parameters as before.
+
+We'll use a different prompt from the one in the config
+
+.. code-block:: bash
+
+    tune run generate --config ./custom_generation_config.yaml \
+    prompt="What are some interesting sites to visit in the Bay Area?"
+
+
+Once generation is complete, you'll see the following in the logs.
+
+
+.. code-block:: bash
+
+    [generate.py:92] A park in San Francisco that sits at the top of a big hill.
+                     There are lots of trees and a beautiful view of San Francisco...
+
+    [generate.py:96] Time for inference: 4.13 sec total, 72.62 tokens/sec
+    [generate.py:99] Memory used: 17.85 GB
+
+With quantization (and torch compile under the hood), we've sped up generation
+by almost 3x!
+
+|
+
+Using TorchTune Checkpoints with other libraries
+------------------------------------------------
+
+As we mentioned above, one of the benefits of handling of the checkpoint
+conversion is that users can directly work with standard formats. This helps
+with interoperability with other libraries since TorchTune doesn't add yet
+another format to the mix.
+
+Let's take a look at an example of how this would work with a popular codebase
+used for running performant inference with LLMs -
+`gpt-fast <https://github.com/pytorch-labs/gpt-fast/tree/main>`_. This section
+assumes that you've cloned the repository on your machine.
+
+``gpt-fast`` makes some assumptions about the checkpoint and the availability of
+the key-to-file mapping. Let's satisfy these assumptions, by creating this mapping
+file. Let's assume we'll be using ``<new_dir>/Llama-2-7B-hf`` as the directory
+for this. ``gpt-fast`` assumes that the directory with checkpoints has the
+same format at the HF repo-id.
+
+.. code-block:: python
+
+    import torch
+
+    # create the output dictionary
+    output_dict = {"weight_map": {}}
+
+    # Load the checkpoints
+    sd_1 = torch.load('/tmp/Llama-2-7b-hf/hf_model_0001_0.pt', mmap=True, map_locations='cpu')
+    sd_2 = torch.load('/tmp/Llama-2-7b-hf/hf_model_0002_0.pt', mmap=True, map_location='cpu')
```

Comment:
Should the folder here be <checkpoint_dir>? (I might just be misunderstanding)

Reply:
nope you're right! good catch